reactioncommerce/example-storefront
# Fixes: handle OAUTH2_IDP_HOST_URL with no ending slash and missing shop

Resolves #626 
Impact: **minor**
Type: **bugfix**

## Changes
- When `OAUTH2_IDP_HOST_URL` env does not end in `/`, logout now works correctly.
- Throughout the app, components have been updated to render as much as possible without needing a shop loaded. The net result is that the UI loads properly and things like login/logout work even when there is no primary shop for the environment yet.

## Breaking changes
None

## Testing
- Verify that logout works when `OAUTH2_IDP_HOST_URL` ends with `/` AND when it does not.
- Connect to an API with a fresh database that has no shops yet. Verify the UI loads and login/logout works.
